src/storage/src/source/source_reader_pipeline.rs

```diff
@@ -419,17 +410,16 @@ where
                     let mut output = output.activate();
                     output.session(&cap).give_container(data);
 
-                    for statuses in statuses_by_idx.values_mut() {
-                        if statuses.is_empty() {
-                            continue;
-                        }
-                        health_output.session(&health_cap).give_container(statuses);
-                        statuses.clear()
+                    if !statuses.is_empty() {
+                        health_output
+                            .session(&health_cap)
+                            .give_container(&mut statuses);
+                        statuses.clear();
```

Comment:
You could just write the status directly I think, without going through the `statuses` vector. That way you'd actually construct even-sized containers provided by the Timely infrastructure.

Reply:
Good point, fixed in the top commit of the PR